File: proton-sdk/client/cs/src/Proton.Drive.Sdk/Nodes/OwnedBy.cs
namespace Proton.Drive.Sdk.Nodes;

/// <summary>
/// Owner of the node (who owns the volume where the node is located).
/// </summary>
/// <param name="Email">Email of the owner for regular and photo volumes, null otherwise.</param>
/// <param name="Organization">Organization name for org. volumes, null otherwise.</param>
public sealed record OwnedBy(string? Email = null, string? Organization = null);

File: proton-sdk/incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/OwnedBy.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.entity.OwnedBy
import proton.drive.sdk.ProtonDriveSdk

fun ProtonDriveSdk.OwnedBy.toEntity() = OwnedBy(
    email = takeIf { hasEmail() }?.email,
    organization = takeIf { hasOrganization() }?.organization,
)
